src/perform_step/low_storage_rk_perform_step.jl

```diff
@@ -182,3 +182,90 @@ end
 
   f(k, u, p, t+dt)
 end
+
+
+
+# 2R+ low storage methods
+function initialize!(integrator,cache::LowStorageRK2RPConstantCache)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+  integrator.kshortsize = 1
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPConstantCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack A₁,Bₗ,B̂ₗ,Bᵢ,B̂ᵢ,Cᵢ = cache
+
+  Xᵢ  = uprev
+  k   = fsalfirst
+  tmp = zero(uprev)
+
+  #stages 1 to s-1
+  for i in eachindex(A₁)
+    tmp = tmp + (Bᵢ[i] - B̂ᵢ[i])*dt*k
+    u   = Xᵢ + A₁[i]*dt*k
+    Xᵢ  = u + (Bᵢ[i] - A₁[i])*dt*k
+    k   = f(u, p, t + Cᵢ[i]*dt)
+  end
+
+  #last stage
+  tmp = tmp + (Bₗ - B̂ₗ)*dt*k
+  u   = Xᵢ  + Bₗ*dt*k
+
+  #Error estimate
+  if integrator.opts.adaptive
+    atmp = calculate_residuals(tmp, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm, t)
+    integrator.EEst = integrator.opts.internalnorm(atmp,t)
+  end
+
+  integrator.k[1] = integrator.fsalfirst
+  integrator.fsallast = f(u, p, t+dt) # For interpolation, then FSAL'd
+  integrator.u = u
+end
+
+function initialize!(integrator,cache::LowStorageRK2RPCache)
+  integrator.kshortsize = 2
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.fsalfirst = cache.fsalfirst
+  integrator.fsallast = cache.k
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst, integrator.uprev, integrator.p, integrator.t)
+end
+
+@muladd function perform_step!(integrator,cache::LowStorageRK2RPCache,repeat_step=false)
+  @unpack t,dt,u,uprev,f,fsalfirst,p = integrator
+  @unpack k,tmp,atmp,Xᵢ = cache
+  @unpack A₁,Bₗ,B̂ₗ,Bᵢ,B̂ᵢ,Cᵢ = cache.tab
+
+  @. Xᵢ = uprev
+  @. k  = fsalfirst
+  @. tmp = zero(u)
+
+  #stages 1 to s-1
+  for i in eachindex(A₁)
+    @. tmp = tmp + (Bᵢ[i] - B̂ᵢ[i])*dt*k
+    @. u   = Xᵢ + A₁[i]*dt*k
+    @. Xᵢ  = u + (Bᵢ[i] - A₁[i])*dt*k
+    f(k, u, p, t + Cᵢ[i]*dt)
+  end
```

Comment:
This part is a bit confusing for me because
- `A₁[i]` is added and subtracted again to get `Xᵢ` (which might add additional numerical errors).
- `Xᵢ` seems to store what will later be `u` while `u` stores intermediate values.

I know, the description in the artcile of Kennedy et al. is also a bit confusing because of the swapping of the registers. But there are nicer (for me) descriptions of these schemes, e.g. Algorithm 2 of [Ketcheson (2010)](http://www.davidketcheson.info/assets/papers/2010_LSRK_postprint.pdf). The adaptive part is missing there but can be added similar to what you did in this loop.

Reply:
I meant to say that this one hasn't been done yet.

Reply:
It's Done!